### PR TITLE
Add Teleport Connect to Headless docs

### PR DESCRIPTION
**Pull request by Joerger**
Adds a section on using Teleport Connect for headless approval.

Depends on https://github.com/gravitational/teleport/pull/29875

### CODE REVIEW

File: docs/pages/access-controls/guides/headless.mdx [hunk 1]
```diff
@@ -178,6 +179,26 @@ alice@server01 $
   your local terminal.
 </Notice>
 
+## Optional: Teleport Connect
+
+Teleport Connect v13.3.1+ can also be used to approve headless logins seamlessly.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/headless.mdx:184]:
```suggestion
Teleport Connect v13.3.1+ can also be used to approve headless logins.
```

File: docs/pages/access-controls/guides/headless.mdx [hunk 2]
```diff
@@ -178,6 +179,26 @@ alice@server01 $
...
+Teleport Connect will automatically detect the headless login attempt and display
+the details in the App for you to `Approve` or `Cancel`.
```
Comment (zmb3) [inline docs/pages/access-controls/guides/headless.mdx:186]:
```suggestion
Teleport Connect will automatically detect the headless login attempt and allow you to approve or cancel the request.
```

File: docs/pages/access-controls/guides/headless.mdx [hunk 3]
```diff
@@ -178,6 +179,26 @@ alice@server01 $
...
+
+<Figure width="700">
+![Headless Confirmation](../../../img/headless/confirmation.png)
+</Figure>
+
+When you hit `Approve`, you will be prompted to tap your MFA key to complete the approval
```
Comment (zmb3) [inline docs/pages/access-controls/guides/headless.mdx:192]: I think we're overusing `Code Formatting`. Approve is just a verb, we don't need any special formatting.

```suggestion
You will be prompted to tap your MFA key to complete the approval
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/access-controls/guides/headless.mdx [hunk 4]
```diff
@@ -35,6 +35,7 @@ For example:
 - Machines for Headless WebAuthn activities have [Linux](../../installation.mdx#linux), [macOS](../../installation.mdx#macos) or [Windows](../../installation.mdx#windows-tsh-client-only) `tsh` binary v12.2+ installed.
 - Machines used to approve Headless WebAuthn requests have a Web browser with [WebAuthn support](
   https://developers.yubico.com/WebAuthn/WebAuthn_Browser_Support/) or `tsh` binary v12.2+ installed.
+- Optional: Teleport Connect v13.3.1+ for [seamless headless approval](#optional-teleport-connect).
```
Comment (ptgott) [inline docs/pages/access-controls/guides/headless.mdx:38]:
```suggestion
- Optional: Teleport Connect v13.3.1+ for [seamless Headless WebAuthn approval](#optional-teleport-connect).
```
We should give this feature a consistent name within this page

File: docs/pages/access-controls/guides/headless.mdx [hunk 5]
```diff
@@ -178,6 +179,25 @@ alice@server01 $
...
+
+You will be prompted to tap your MFA key to complete the approval process.
+
+<Figure width="700">
+![Headless WebAuthn Approval](../../../img/headless/approval.png)
+</Figure>
+
+<Notice type="note">
+  This also requires a v13.3.1+ Teleport Auth server.
+</Notice>
```
Comment (ptgott) [inline docs/pages/access-controls/guides/headless.mdx:200]:
```suggestion
Teleport Connect v13.3.1+ can also be used to approve Headless WebAuthn logins.
Teleport Connect will automatically detect the Headless WebAuthn login attempt
and allow you to approve or cancel the request.

<Figure width="700">
![Headless Confirmation](../../../img/headless/confirmation.png)
</Figure>

You will be prompted to tap your MFA key to complete the approval process.

<Figure width="700">
![Headless WebAuthn Approval](../../../img/headless/approval.png)
</Figure>

<Notice type="note">
  This also requires a v13.3.1+ Teleport Auth Service.
</Notice>
```

So we have consistent product names in the docs.

File: docs/pages/connect-your-client/teleport-connect.mdx [hunk 6]
```diff
@@ -212,6 +212,7 @@ Below is the list of the supported config properties.
 | `keymap.openClusters`         | `Command+E` on macOS<br/>`Ctrl+E` on Windows/Linux                                                                   | Shortcut to open the cluster selector.                                 |
 | `keymap.openProfiles`         | `Command+I` on macOS<br/>`Ctrl+I` on Windows/Linux                                                                   | Shortcut to open the profile selector.                                 |
 | `keymap.openSearchBar`        | `Command+K` on macOS<br/>`Ctrl+K` on Windows/Linux                                                                   | Shortcut to open the search bar.                                       |
+| `headless.skipConfirm` | false | Skips the confirmation prompt for headless login approval and instead prompts for WebAuthn immediately. |
```
Comment (ptgott) [inline docs/pages/connect-your-client/teleport-connect.mdx:215]:
```suggestion
| `headless.skipConfirm` | false | Skips the confirmation prompt for Headless WebAuthn login approval and instead prompts for WebAuthn immediately. |
```
For consistent naming

**Review by ptgott — APPROVED** [review]
Some minor suggestions for consistent naming of Teleport products

**Pull request merged**